### PR TITLE
Spring cache backed Token store 

### PR DESCRIPTION
**Pull request by tgkprog**
Spring cache backed Token store tested with ConcurrentMapCache and Spring redis data cache.  Can test with ConcurrentMapCache by changing in cache.properties 

```
#1 : simple in mem. 2 : Redis or class name
cache.use=2

#1 : in mem token store. 2 : cache backed store or class name
store.use=2
#if redis
redis.host-name=localhost
redis.port=6379
```

### CODE REVIEW

**Comment by dsyer** [comment]
Interesting idea, and since it only affects a sample it might be worth considering with some changes. Did you fill in the contributor's agreement (link in README)?

**Comment by tgkprog** [comment]
I'm open to making changes (that you suggest) or you can if its easier. I will sign it, its open source.

**Comment by dsyer** [comment]
OK, so now I looked at your code, I'm not sure I like the `TokenStore` implementation. Isn't the point of a cache that it *isn't* a repository? I would expect that maybe we can decorate the existing `TokenStore` with a caching interceptor though (and then choose the cache implementation there).

**Comment by tgkprog** [comment]
I see cache as a repo, just in the RAM, in fact can have a MySql Db all in the RAM if you want to (and sync to a file db for persistance), but that is off topic. But the question for me was how to implement TokenStore. I choose this metod due to code similar to below: 

````java
public void storeAccessToken(OAuth2AccessToken token, OAuth2Authentication authentication) {
		if (this.flushCounter.incrementAndGet() >= this.flushInterval) {
			flush();
			this.flushCounter.set(0);
		}
		this.accessTokenStore.put(token.getValue(), token);
		this.authenticationStore.put(token.getValue(), authentication);
		this.authenticationToAccessTokenStore.put(authenticationKeyGenerator.extractKey(authentication), token);
		if (!authentication.isClientOnly()) {
			addToCollection(this.userNameToAccessTokenStore, getApprovalKey(authentication), token);
		}
````

And

````java
public void removeAccessToken(String tokenValue) {
		OAuth2AccessToken removed = this.accessTokenStore.remove(tokenValue);
		this.accessTokenToRefreshTokenStore.remove(tokenValue);
		// Don't remove the refresh token - it's up to the caller to do that
		OAuth2Authentication authentication = this.authenticationStore.remove(tokenValue);
		if (authentication != null) {
			this.authenticationToAccessTokenStore.remove(authenticationKeyGenerator.extractKey(authentication));
			Collection<OAuth2AccessToken> tokens;
			String clientId = authentication.getOAuth2Request().getClientId();
			tokens = this.userNameToAccessTokenStore.get(getApprovalKey(clientId, authentication.getName()));
			if (tokens != null) {
				tokens.remove(removed);
			}
````

I guess can use annotations but would be complex to rework for me & more time than I can give . Let me know if you want me to fix few and i can spend 4-5 hours or reject this approach?

**Comment by pivotal-issuemaster** [comment]
@tgkprog Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-security-oauth&pullRequestId=465) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@tgkprog Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security-oauth&pullRequestId=465)!

**Comment by tgkprog** [comment]
Should I cancel this pull as it seems to be rejected ?

**Comment by dsyer** [comment]
I was happy to consider modifications. But if you want to close it and send another one that's fine too.

File: samples/oauth2/sparklr/pom.xml [hunk 1]
```diff
@@ -1,4 +1,5 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```
Comment (tgkprog) [inline samples/oauth2/sparklr/pom.xml]: test

Comment (sajithmx) [inline samples/oauth2/sparklr/pom.xml]: yes

File: samples/oauth2/sparklr/pom.xml [hunk 2]
```diff
@@ -14,6 +15,11 @@
 	<name>OAuth for Spring Security - Sparklr2 (OAuth 2 Provider Example)</name>
```
Comment (tgkprog) [inline samples/oauth2/sparklr/pom.xml]: Code checker

**Comment by jgrandja** [comment]
@tgkprog As you may be aware, this project is scheduled to reach EOL. Please see the [announcement](https://spring.io/blog/2020/05/07/end-of-life-for-spring-security-oauth) for further details.

I'm going to close this PR as it doesn't make sense to merge these changes given the status of the project.

**Pull request closed without merging**